ddtrace/opentelemetry/tracer.go

```diff
@@ -52,15 +51,21 @@ func (t *oteltracer) Start(ctx context.Context, spanName string, opts ...oteltra
 		ddopts = append(ddopts, tracer.Tag(ext.SpanKind, k.String()))
 	}
 	telemetry.GlobalClient.Count(telemetry.NamespaceTracers, "spans_created", 1.0, telemetryTags, true)
-	ctxOpts, ok := spanOptionsFromContext(ctx)
-	if ok {
-		ddopts = append(ddopts, ctxOpts...)
-	}
 	var cfg ddtrace.StartSpanConfig
-	for _, option := range ctxOpts {
-		option(&cfg)
+	if opts, ok := spanOptionsFromContext(ctx); ok {
+		ddopts = append(ddopts, opts...)
+		for _, o := range opts {
+			o(&cfg)
+		}
+	}
+	finalTags := make(map[string]interface{})
```

Comment:
It'd be simpler and be one fewer allocation if you just re-used the `cfg.Tags`. Unless there's a reason we can't? e.g. like this:

replace 61-68:
```
	for _, attr := range ssConfig.Attributes() {
		cfg.Tags[string(attr.Key)] = attr.Value.AsInterface()
	}
```

and then just use `cfg.Tags` as the `attributes` value when constructing the `oteltrace.Span` below.